src/pages/home/report/ReportActionItem.js

```diff
@@ -196,7 +196,12 @@ class ReportActionItem extends Component {
                                 <OfflineWithFeedback
                                     onClose={() => {
                                         if (this.props.action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD) {
-                                            ReportActions.deleteOptimisticReportAction(this.props.report.reportID, this.props.action.clientID);
+                                            const sequenceNumber = this.props.action.actionName
+                                              === CONST.REPORT.ACTIONS.TYPE.IOU
+                                                ? this.props.action
+                                                    .sequenceNumber
+                                                : this.props.action.clientID;
+                                            ReportActions.deleteOptimisticReportAction(this.props.report.reportID, sequenceNumber);
```

Comment:
Can we just use the `clientID` instead of `sequenceNumber` so we don't need this logic?

Reply:
When the action is of type IOU (, I found that the `clientID` is a random generated ID and not a `sequenceNumber`. So when an error occurs, clicking on the `X` does not delete the sendMoney action, because there isn't any action keyed by that ID. How do you suggest we fix this?

Reply:
Here's where it's happening: https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.js#L795

Reply:
Sorry, I completely missed this. I think that keeping it until `sequenceNumbers` are deprecated sounds good.